### PR TITLE
API call should call from the correct model

### PR DESCRIPTION
**Pull request by Nolle10**

### CODE REVIEW

File: CarboPlanner/src/views/Loading.tsx [hunk 1]
```diff
@@ -28,16 +199,24 @@ export default function Loading({route}) {
...
         const requestBody = {
-          image: firstImageBase64
+          image: 
+            allImagesBase64.map((uri, index) => ({"id": index, "img": uri}))
         };
-    
+
         //Sends POST request to this URL, sends using the Base64 Image from the Expo Camera
-        fetch(api_url, {
+        fetch(asd, {
```
Comment (Nolle10) [inline CarboPlanner/src/views/Loading.tsx]:
```suggestion
        fetch(myNewFerrari, {
```

File: CarboPlanner/src/views/Loading.tsx [hunk 2]
```diff
@@ -54,7 +233,12 @@ export default function Loading({route}) {
         })
         .then(function(data) {
           // The JSON data from the WebServer is returned here
-            navigation.navigate('Result', {base64: firstImageBase64, data: data, allImages: allImagesBase64});
+
+            console.log(data);
```
Comment (Nolle10) [inline CarboPlanner/src/views/Loading.tsx]:
```suggestion
```

File: CarboPlanner/src/views/TrackingScreen.tsx [hunk 3]
```diff
@@ -53,17 +53,42 @@ export default function TrackingScreen(){
...
+        setProtein(protein)
+        setCarbs(carbs)
+        setFat(fat)
+        setCalories(calories)
+
+        return {protein, carbs, fat, calories}
+    };
 
     const renderMealItem = ({ item }) => {
+        console.log(item, "hello item")
```
Comment (Nolle10) [inline CarboPlanner/src/views/TrackingScreen.tsx:88]:
```suggestion
```

**Review by Nolle10 — COMMENTED** [review]
this should be changed

File: CarboPlanner/src/components/List.tsx [hunk 4]
```diff
@@ -19,7 +19,7 @@ export default function List(props: any) {
             <View style={styles.titleRow}>
                 <Text style={styles.bold}>{props.name}</Text>
                 {/*props.showOptions determines whether the <OptionButton /> component should be rendered or not */}
-                {props.showOptions && <OptionButton uuidKey={props.uuidKey}/>}
+                {props.showOHptions && <OptionButton uuidKey={props.uuidKey}/>}
```
Comment (Sindahlof) [inline CarboPlanner/src/components/List.tsx]: Why is it change to showOHptions

File: CarboPlanner/src/views/Loading.tsx [hunk 5]
```diff
@@ -184,17 +175,20 @@ export default function Loading({route}) {
 
     //const api_url = process.env.EXPO_PUBLIC_API_URL;
```
Comment (Sindahlof) [inline CarboPlanner/src/views/Loading.tsx]:
```suggestion
    const api_url = process.env.EXPO_PUBLIC_API_URL;
```

Please change it back to using the .env file

File: CarboPlanner/src/views/Loading.tsx [hunk 6]
```diff
@@ -44,22 +44,13 @@ export default function Loading({route}) {
                 const volume = getVolume(height, width, sideWidth, reference); //cm^3
                 // Store the calculated volume in the listOfVolumes map
                 listOfVolumes.set(name, volume);
+                //console.log(`Volume of ${name}:`, volume);
```
Comment (frmad) [inline CarboPlanner/src/views/Loading.tsx]:
```suggestion
               
```

File: CarboPlanner/src/views/Loading.tsx [hunk 7]
```diff
@@ -184,17 +175,20 @@ export default function Loading({route}) {
 
     //const api_url = process.env.EXPO_PUBLIC_API_URL;
 
+    const myNewFerrari : string = 'https://testapi-5qhj5kt2ta-lz.a.run.app/v1/object-detection';
+
     const api_url : string = 'https://yolov5-flaskapi-5qhj5kt2ta-lz.a.run.app/v1/object-detection';
     //GAMLE - https://yolov5-flaskapi-5qhj5kt2ta-lz.a.run.app/v1/object-detection
```
Comment (frmad) [inline CarboPlanner/src/views/Loading.tsx]:
```suggestion
 
```

File: CarboPlanner/src/views/Loading.tsx [hunk 8]
```diff
@@ -184,17 +175,20 @@ export default function Loading({route}) {
...
 
         const requestBody = {
-          image: firstImageBase64
+          image: 
+            allImagesBase64.map((uri, index) => ({"id": index, "img": uri}))
         };
 
         //Sends POST request to this URL, sends using the Base64 Image from the Expo Camera
-        fetch(api_url, {
+        fetch(myNewFerrari, {
```
Comment (frmad) [inline CarboPlanner/src/views/Loading.tsx]: change the name
```suggestion
        fetch(api_url_new, {
```

**Review by frmad — DISMISSED** [review]
small issues

**Review by Sindahlof — APPROVED** [review]
Looks good now

**Pull request merged**